public/plugin.ts

```diff
@@ -29,15 +29,27 @@ import {
 import { LOGIN_PAGE_URI, PLUGIN_NAME } from '../common';
 import { API_ENDPOINT_PERMISSIONS_INFO } from './apps/configuration/constants';
 
+async function getApiPermission(core: CoreSetup): Promise<boolean | undefined> {
+  try {
+    const permissions = await core.http.get(API_ENDPOINT_PERMISSIONS_INFO);
+    return permissions?.data?.has_api_access;
```

Comment:
Minor: we can do `|| false` so the return type could be just `Promise<boolean>`

Reply:
Added || false. But tslint check shows that undefined is still needed because of the exception throwing.